bulk-dqm/bulk-data-export-client
# Set all default options

# Summary
The client now sets default values for all the configurable client options.

## Code changes
* Adds`config/defaults.js` - this is modeled off the SMART bulk-data-client’s [default config file](https://github.com/smart-on-fhir/bulk-data-client/blob/main/config/defaults.js) but has additional fields for options specific to our client (such as the measure bundle and its associate output path for outputting the FHIR MeasureReports). 
* Various `cli.ts` updates:
    * Imports the `TextReporter` so it can now be used as an alternative to the `CLIReporter`
    * Removes defaults that were set within the CLI, like `inlineDocRefAttachmentTypes`, since they are now set in the default config file
    * Adds code to set the default options along with the options set via the CLI and those specified by a custom config file (if specified with the `—config` flag)
    * Switched `requiredOption` to `option` for required options, since they no longer need to be specified from the CLI - they can be specified by the custom config file now. The defaults file sets these required options to `null` by default so that errors are still thrown if they are missing

## Testing Guidance
Test several CLI commands. Some examples include:
* Set all the options entirely with the CLI
* Set all the options with a custom config file and use the `—config` flag in the CLI to point to the file
* Set some of the options in the custom config file, and the rest directly via the CLI
* Check that existing CLI logic still holds (still throws error for missing private key when token URL specified, logging still works, etc.)
